cli/command/context/create_test.go

```diff
@@ -0,0 +1,230 @@
+		inStream := command.NewInStream(pipeReader)
+		outStream := command.NewOutStream(pipeWriter)
+		m.in = func() *command.InStream { return inStream }
+		m.out = func() *command.OutStream { return outStream }
+		go func() {
+			<-close
+			pipeWriter.Close()
+		}()
+	}
+}
+
+func withOut(w io.Writer) func(*cliMock) {
+	return func(m *cliMock) {
+		outStream := command.NewOutStream(w)
+		m.out = func() *command.OutStream { return outStream }
+	}
+}
+
+func TestCreateNoName(t *testing.T) {
+	cli, cleanup, err := makeCliMock(t)
+	assert.NilError(t, err)
+	defer cleanup()
+	err = runCreate(cli, &createOptions{})
+	assert.ErrorContains(t, err, "context name is required")
+}
+
+func TestCreateExitingContext(t *testing.T) {
+	cli, cleanup, err := makeCliMock(t)
+	assert.NilError(t, err)
+	defer cleanup()
+	assert.NilError(t, cli.ContextStore().CreateOrUpdateContext("test", store.ContextMetadata{}))
+
+	err = runCreate(cli, &createOptions{
+		name: "test",
+	})
+	assert.ErrorContains(t, err, "context \"test\" already exists")
+}
+
+func TestCreateInvalidOrchestrator(t *testing.T) {
+	cli, cleanup, err := makeCliMock(t)
+	assert.NilError(t, err)
+	defer cleanup()
+
+	err = runCreate(cli, &createOptions{
+		name:                     "test",
+		defaultStackOrchestrator: "invalid",
+	})
+	assert.ErrorContains(t, err, "specified orchestrator \"invalid\" is invalid, please use either kubernetes, swarm or all")
+}
+
+func TestCreateOrchestratorSwarm(t *testing.T) {
+	cli, cleanup, err := makeCliMock(t)
+	assert.NilError(t, err)
+	defer cleanup()
+
+	err = runCreate(cli, &createOptions{
+		name:                     "test",
+		defaultStackOrchestrator: "swarm",
+	})
+	assert.NilError(t, err)
+}
+
+func TestCreateOrchestratorEmpty(t *testing.T) {
+	cli, cleanup, err := makeCliMock(t)
+	assert.NilError(t, err)
+	defer cleanup()
+
+	err = runCreate(cli, &createOptions{
+		name: "test",
+	})
+	assert.NilError(t, err)
+}
+
+func TestCreateUse(t *testing.T) {
+	configDir, err := ioutil.TempDir("", t.Name()+"config")
+	assert.NilError(t, err)
+	defer os.RemoveAll(configDir)
+	configFilePath := filepath.Join(configDir, "config.json")
+	testCfg := configfile.New(configFilePath)
+	cli, cleanup, err := makeCliMock(t, withCliConfig(testCfg))
+	assert.NilError(t, err)
+	defer cleanup()
+
+	err = runCreate(cli, &createOptions{
+		name: "test",
+		use:  true,
+	})
+	assert.NilError(t, err)
+	reloadedConfig, err := config.Load(configDir)
+	assert.NilError(t, err)
+	assert.Equal(t, "test", reloadedConfig.CurrentContext)
+}
+
+func TestCreateOrchestratorKubernetesNoEndpoint(t *testing.T) {
+	cli, cleanup, err := makeCliMock(t)
+	assert.NilError(t, err)
+	defer cleanup()
+
+	err = runCreate(cli, &createOptions{
+		name:                     "test",
+		defaultStackOrchestrator: "kubernetes",
+	})
+	assert.ErrorContains(t, err, "cannot specify orchestrator \"kubernetes\" without configuring a Kubernetes endpoint")
+}
+func TestCreateOrchestratorAllNoKubernetesEndpoint(t *testing.T) {
```

Comment:
I guess this test is a little too much, as it only tests that `all` value `HasKubernetes`.

Reply:
I wanted to make sure we use HasKubernetes in the callee and not just an equality comparison with orchestratorKubernetes